routing/router.go

```diff
@@ -180,7 +179,15 @@ type Config struct {
 	SendToSwitch func(firstHop lnwire.ShortChannelID,
 		paymentID uint64,
 		htlcAdd *lnwire.UpdateAddHTLC,
-		circuit *sphinx.Circuit) ([sha256.Size]byte, error)
+		circuit *sphinx.Circuit) error
+
+	// GetPaymentResult returns the the result of the payment attempt with
+	// the given paymentID. The method returns a channel where the payment
+	// result will be sent when available, or an error is encountered. If
+	// the paymentID is unknown, htlcswitch.ErrPaymentIDNotFound will be
+	// returned.
+	GetPaymentResult func(paymentID uint64) (
```

Comment:
Not a blocker, but at this point it may make sense to extract this method and the one above it into a new interface.

Reply:
Made interface